Scripts/UI/Base/CircleMenuButton.gd

```diff
@@ -0,0 +1,113 @@
+@tool
+extends BaseButton
+class_name CircleMenuButton
+## A custom radial selection menu for displaying and interacting with options within a 
+## circular interface.
+
+# --- Constants ---
+const SPRITE_SIZE = Vector2(32, 32)                    ## Size for aligning option text
+
+# --- Exported Properties ---
+@export var options: Array[WheelOption]                ## List of selectable options
+
+@export_group("Color", "color_")
+@export var color_bg: Color = Color(0.3, 0.3, 0.3)     ## Background circle color
+@export var color_line: Color = Color(1, 1, 1)         ## Line color for option dividers
+@export var color_highlight: Color = Color(1, 0.5, 0)  ## Highlight color for selected option
+
+@export_group("size", "size_")
+@export var size_outer_radius: int = 256               ## Radius for the outer circle
+@export var size_inner_radius: int = 64                ## Radius for the inner circle
+@export var size_line_width: int = 4                   ## Width of the dividing lines
+
+
+# --- Private Properties ---
+var _selection: int = -1  # Currently selected option index
+
+# --- Signals ---
+signal closed ## Signal emitted when the _selection is confirmed
+
+# --- Built-in Callbacks ---
+func _process(_delta: float) -> void:
+	if Engine.is_editor_hint():
+		return
+	if !visible:
+		return
+	var mouse_pos = get_local_mouse_position()
+	var mouse_radius = mouse_pos.length()
+
+	if mouse_radius < size_inner_radius:
+		_selection = -1  # Set _selection to -1 if the mouse is inside the inner circle
+	else:
+		var mouse_angle = mouse_pos.angle()
+		_selection = int(fposmod(mouse_angle, TAU) / (TAU / options.size()))  # Calculate based on angle
+
+	if _selection >= options.size():
+		_selection = options.size() - 1
+	elif _selection < 0:
+		_selection = -1
+
+	if Input.is_action_just_pressed("ui_accept"):
+		closed.emit()
+
+	queue_redraw()
+
+func _unhandled_input(event: InputEvent) -> void:
+	if not Engine.is_editor_hint():
+		if event is InputEventMouseButton and event.button_index == MOUSE_BUTTON_LEFT and event.pressed and visible:
+			get_tree().get_root().set_input_as_handled()
+			closed.emit()
+
+func _draw() -> void:
+	var offset = SPRITE_SIZE / -2  # Offset for centering text on options
```

Comment:
This one is mostly good because AFIK it's not super easy to do circles in godot's UI. Plus we probably won't being doing much post-processing on the UI. Amazing implementation.

Reply:
thanks! unfortunately this is only a temporary, until we get a proper graphics designer who will make a nice ui system for the doors.